graphql2/graphqlapp/calendarsubscription.go

```diff
@@ -0,0 +1,100 @@
+package graphqlapp
+
+import (
+	"context"
+	"database/sql"
+	"github.com/target/goalert/calendarsubscription"
+	"github.com/target/goalert/graphql2"
+	"github.com/target/goalert/search"
+)
+
+func (q *Query) CalendarSubscription(ctx context.Context, id string) (*calendarsubscription.CalendarSubscription, error) {
+	return q.CalendarSubscriptionStore.FindOne(ctx, id)
+}
+
+// todo: return url instead of bool once endpoint has been created
+func (m *Mutation) CreateCalendarSubscription(ctx context.Context, input graphql2.CreateCalendarSubscriptionInput) (res bool, err error) {
+	err = withContextTx(ctx, m.DB, func(ctx context.Context, tx *sql.Tx) error {
+		cs := &calendarsubscription.CalendarSubscription{
+			Name: input.Name,
+		}
+
+		res, err = m.CalendarSubscriptionStore.CreateSubscriptionTx(ctx, tx, cs)
+		if err != nil {
+			return err
+		}
+
+		// todo: gen url for user
+
+		return err
+	})
+
+	return res, err
+}
+
+func (m *Mutation) UpdateCalendarSubscription(ctx context.Context, input graphql2.UpdateCalendarSubscriptionInput) (bool, error) {
+	err := withContextTx(ctx, m.DB, func(ctx context.Context, tx *sql.Tx) error {
+		cs := &calendarsubscription.CalendarSubscription{
+			ID:       input.ID,
+			Name:     *input.Name,
+			Disabled: *input.Disabled,
+		}
+
+		err := m.CalendarSubscriptionStore.UpdateSubscriptionTx(ctx, tx, cs)
+		if err != nil {
+			return err
+		}
+
+		return err
+	})
+
+	return err == nil, err
+}
+func (q *Query) CalendarSubscriptions(ctx context.Context, opts *graphql2.CalendarSubscriptionSearchOptions) (conn *graphql2.CalendarSubscriptionConnection, err error) {
```

Comment:
This really should just be an array using a `FindAllByUser` approach -- the result set will always be scoped to a specific user like contact methods (or like integration keys on a service).

It's a lot of overhead to add/maintain otherwise, search would generally be for things created by other users and/or >100 results.

Here's what we had settled on previously: https://github.com/target/goalert/compare/calendar-subscription-api#diff-04e3af9ffc876a383b6f192a8ff8450dR841